bofire/benchmarks/aspen_benchmark.py

```diff
@@ -0,0 +1,199 @@
+import logging
+import os
+from typing import Callable, Optional
+
+import pandas as pd
+
+from bofire.benchmarks.benchmark import Benchmark
+from bofire.domain import Domain
+
+# Create a folder for the log file, if not alredy exists.
+if not os.path.exists("bofire_logs"):
+    os.makedirs("bofire_logs")
+
+# Create a Logger that safes console output.
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s:%(levelname)s:%(message)s")
+file_handler = logging.FileHandler("bofire_logs/aspen_benchmark.log")
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+
+class Aspen_benchmark(Benchmark):
+    """This class connects to a Aspen plus file that runs the desired process.
+    It writes incoming input values into Aspen plus, runs the simulation and returns the results.
+    When initializing this class, make sure not to block multiple Aspen plus licenses at once
+    when is not absolutely needed.
+
+    Args:
+        Benchmark: Subclass of the Benchmark function class.
+    """
+
+    def __init__(
+        self,
+        filename: str,
+        domain: Domain,
+        paths: dict[str, str],
+        additional_outputs: Optional[list] = [],
+        translate_into_aspen_readable: Optional[
+            Callable[[Domain], pd.DataFrame]
+        ] = None,
+    ) -> None:
+        """Initializes Aspen_benchmark. A class that connects to Aspen plus.
+
+        Args:
+            filename (str): Filepath of the Aspen plus simulation file.
+            domain (Domain): Domain of the benchmark setting inclunding bounds and information about input values.
+            paths (dict[str, str]): A dictionary with the key value pairs "key_of_variable": "path_to_variable".
+            The keys must be the same as provided in the domain.
+            translate_into_aspen_readable (Optional: Callable): A function that converts dataframe columns into
+            integers or floats so Aspen plus is able to read their values.
+
+        Raises:
+            ValueError: In case the number of provided variable names does not match the number of provided Aspen variable tree paths.
+        """
+        if os.path.exists(filename):
+            self.filename = filename
+        else:
+            raise ValueError("Unable to find Aspen file " + filename)
+
+        self.translate_into_aspen_readable = translate_into_aspen_readable
+        self._domain = domain
+        # Get the variable names (keys) from the domain to access them later easily.
+        self.keys = [self.domain.inputs.get_keys(), self.domain.outputs.get_keys()]
+        self.input_keys = self.domain.inputs.get_keys()
+        self.output_keys = self.domain.outputs.get_keys()
+        self.additional_outputs = []
+        if additional_outputs is not None:
+            self.additional_outputs = additional_outputs
+
+        for key_list in [self.input_keys, self.output_keys]:
+            for key in key_list:
```

Comment:
```suggestion
        for key in self.domain.get_feature_keys():
```

And delete the line above. In this way you can just loop all feature keys in the domain.